File: Wirtualny świat/C++/ConsoleApplication18/robak.cpp
#include "stdafx.h"
#include "robak.h"


robak::robak(swiat& ref, int x, int y) :zwierze(ref, x, y)
{
	this->sila = 1;
	this->inicjatywa = 1;
	this->nazwa = 'R';
	this->specjalna_kolizja = 1;
}
char robak::rysowanie() {
	return this->nazwa;
}
void robak::spec_kolizja(organizm* org) {
	org->ustaw_sila_organizmu(org->sila_organizmu() + 2);
	cout << org->rysowanie() << " zwiekszyl swa sile o 2\n";
}
void robak::nowe(swiat& ref, int x, int y)
{
	organizm *o = new robak(ref, x, y);
	cout << "dodano: " << this->rysowanie() << " na polu: " << x + 1 << ':' << y + 1 << ":\n";
}
robak::~robak()
{
}

File: Wirtualny świat/java/src/organizm/robak.java

package organizm;
import swiat.swiat;


public class robak extends zwierze {
    public robak(swiat WielkiSwiat, int x, int y)
    {
        super(WielkiSwiat, x, y);
      	this.sila = 1;
	this.inicjatywa = 1;
	this.nazwa = 'R';
	this.specjalna_kolizja = true;
    }
    
    @Override
  public void spec_kolizja(organizm org) {
	org.ustaw_sila_organizmu(org.sila_organizmu() + 2);
        System.out.println(org.rysowanie() + " na polu: " + (x+1) + ":" + (y+1) + " zwiekszyl swoja sile o 2");
}


    public void nowy(swiat WielkiSwiat, int x, int y)
    {
        organizm o = new zmija(WielkiSwiat, x, y);
    }

}
